crates/hir_ty/src/lower.rs

```diff
@@ -1177,8 +1188,9 @@ pub enum ValueTyDefId {
     EnumVariantId(EnumVariantId),
     ConstId(ConstId),
     StaticId(StaticId),
+    ConstParamId(ConstParamId),
```

Comment:
hmm it feels a bit weird to add `ConstParamId` here, since a const param is not a "def" in our usual sense. I think it would make more sense to have a separate query `const_param_ty` and just do something like
```rust
            ValueNs::GenericParam(it) => return db.const_param_ty(it),
```
in `resolve_value_path` (in infer/path.rs).
That `const_param_ty` query should also just return `Ty`, not `Binders<Ty>`, because a const param can't have type parameters itself.

Reply:
Now things make more sense, this simplifies this quite a bit